mks/model/stuf_02_04.py

```diff
@@ -314,6 +317,57 @@ def extract_verbintenis_data(persoon_tree: Tag):
 
 def extract_address(persoon_tree: Tag, is_amsterdammer):
     result = []
+    
+    addresses_vbl = format_addresses(persoon_tree.find_all('PRSADRVBL'), ADDR_VBL, is_amsterdammer)
+    addresses_cor = format_addresses(persoon_tree.find_all('PRSADRCOR'), ADDR_COR, is_amsterdammer)
+    addresses = merge_sort_addresses(addresses_vbl, addresses_cor)
+    if is_nil(addresses):
+        return {}, []
+
+    return get_current_past_addresses(addresses)
+    
+def get_current_past_addresses(addresses):
+    current = None
+    past = []   
+  
+    for address in addresses:
+        end = address["einddatumVerblijf"]
+        if current is None and (end is None or end > date.today()):
+            current = address
+        else:
+            if address.get("_adresSleutel"):
+                del address["_adresSleutel"]
+            past.append(address)
+
+    past.sort(key=lambda x: x["einddatumVerblijf"] or date.min, reverse=True)
+
+    # Punt adres is a thing so people are not registered on an address (which has al kinds of implications for that adres)
+    # replace the recent with minimum data and mark it as Adres in onderzoek
+    # see MIJN-2825
+    if current["straatnaam"] == ".":
+        current = {
+            "woonplaatsNaam": "Amsterdam",
+            "straatnaam": ".",
+            "begindatumVerblijf": current["begindatumVerblijf"],
+            "inOnderzoek": True,
+            "landcode": "0000",
+            "landnaam": "Nederland",
+        }
+
+    return current, past    
+
+def merge_sort_addresses(first_group, second_group):
+    all_addresses = first_group + second_group
+    if is_nil(all_addresses):
+        return []
+    all_addresses.sort(key=lambda x: x["begindatumVerblijf"] or date.min, reverse=True)
+    return all_addresses 
+
+def format_addresses(addresses, address_type, is_amsterdammer):
+    result = []
+    if is_nil(addresses):
+        return []
+        
```

Comment:
Dit mag weg.

Reply:
Moet blijven, het kan zijn dat een van de twee calls(vbl/cor) None is, dan moet er een array terug worden gegeven anders crashed de boel.